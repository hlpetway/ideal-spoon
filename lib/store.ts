import type { Action, ThunkAction } from "@reduxjs/toolkit";
import { configureStore } from "@reduxjs/toolkit";
import sharkSliceReducer from "./features/shark/sharkSlice";

export const store = configureStore({
    reducer: {
      shark: sharkSliceReducer
    },
  });

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ThunkReturnType = void> = ThunkAction<
  ThunkReturnType,
  RootState,
  unknown,
  Action
>;
