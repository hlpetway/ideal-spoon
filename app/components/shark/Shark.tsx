"use client";
import { useEffect } from "react";
import { Paper } from "@mui/material";
import {
    selectImage,
    selectSubTitle,
    selectTitle,
    selectSales,
    selectTags
} from "../../../lib/features/shark/sharkSlice";
import { fetchShark } from "../../../lib/features/shark/sharkAPI";
import { useAppDispatch, useAppSelector } from "../../../lib/hooks";
import { SalesTable } from "../SalesTable";
import { Grid } from "@mui/material";
import { Stack } from "@mui/material";
import { Chip } from "@mui/material";
import styles from "../../styles/layout.module.css";

export const Shark = () => {
    const dispatch = useAppDispatch();
    const productImage = useAppSelector(selectImage);
    const productSubTitle = useAppSelector(selectSubTitle);
    const productTitle = useAppSelector(selectTitle);
    const productTags = useAppSelector(selectTags);
    const sharkSales = useAppSelector(selectSales);

    useEffect(() => {
        dispatch(fetchShark());
    },[]);

    return (
        <Grid className={styles.gridContainer} container spacing={2}>
            <Grid item xs={3}>
                <Paper className={styles.paper}  elevation={1}>
                <div className={styles.productDetails}>
                    <img className={styles.productImg} src={productImage} alt={productTitle} />
                    <h2 className={styles.productTitle}>
                        {productTitle}
                    </h2>
                    <h4 className={styles.productSubTitle}>{productSubTitle}</h4>
                    <div className={styles.productTags}>
                        <Stack direction="row" spacing={1}>
                        {productTags?.map((tag) => (
                            <Chip key={tag} label={tag} className={styles.tag}/>
                        ))}
                        </Stack>
                    </div>
                </div>
                </Paper>
            </Grid>
            <Grid item xs={9}>
                <div className="shark-sales-chart">
                    <img className={styles.sharkSalesChart}src="chart.png" alt="a chart of shark sales over time"/>
                    </div>
                <div className="shark-sales">
                    <div className="sales-graph">
                    </div>
                    <div className={styles.salesDetails}>
                        <SalesTable productTitle={productTitle} sales={sharkSales} />
                    </div>
                </div>
            </Grid>
        </Grid >
    );
};
