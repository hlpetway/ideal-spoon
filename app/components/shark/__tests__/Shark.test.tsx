import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import { store } from "../../../../lib/store";
import { Shark } from "../Shark";
const styles = {
    gridContainer: "",
    paper: "",
    productDetails: "",
    productImg: "",
    productTitle: "",
    productSubTitle: "",
    productTags: "",
    tag: "",
    sharkSalesChart: "",
    salesDetails: "",
};

describe("Shark component", () => {
    it("should match the snapshot", () => {
        const component = renderer.create(<Provider store={store}><Shark styles={styles} /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});