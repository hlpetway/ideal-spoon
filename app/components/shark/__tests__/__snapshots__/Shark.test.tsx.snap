// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shark component should match the snapshot 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-mhc70k-MuiGrid-root{box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;width:100%;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;margin-top:-16px;width:calc(100% + 16px);margin-left:-16px;}.css-mhc70k-MuiGrid-root>.MuiGrid-item{padding-top:16px;}.css-mhc70k-MuiGrid-root>.MuiGrid-item{padding-left:16px;}",
      }
    }
    data-emotion="css mhc70k-MuiGrid-root"
  />,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1equabv-MuiGrid-root{box-sizing:border-box;margin:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-basis:25%;-ms-flex-preferred-size:25%;flex-basis:25%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:25%;}@media (min-width:600px){.css-1equabv-MuiGrid-root{-webkit-flex-basis:25%;-ms-flex-preferred-size:25%;flex-basis:25%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:25%;}}@media (min-width:900px){.css-1equabv-MuiGrid-root{-webkit-flex-basis:25%;-ms-flex-preferred-size:25%;flex-basis:25%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:25%;}}@media (min-width:1200px){.css-1equabv-MuiGrid-root{-webkit-flex-basis:25%;-ms-flex-preferred-size:25%;flex-basis:25%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:25%;}}@media (min-width:1536px){.css-1equabv-MuiGrid-root{-webkit-flex-basis:25%;-ms-flex-preferred-size:25%;flex-basis:25%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:25%;}}",
        }
      }
      data-emotion="css 1equabv-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1ps6pg7-MuiPaper-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);}",
          }
        }
        data-emotion="css 1ps6pg7-MuiPaper-root"
      />
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <div
          className=""
        >
          <img
            alt="Loading Product Details"
            className=""
            src=""
          />
          <h2
            className=""
          >
            Loading Product Details
          </h2>
          <h4
            className=""
          />
          <div
            className=""
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-niqf4j-MuiStack-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}.css-niqf4j-MuiStack-root>:not(style):not(style){margin:0;}.css-niqf4j-MuiStack-root>:not(style)~:not(style){margin-left:8px;}",
                }
              }
              data-emotion="css niqf4j-MuiStack-root"
            />
            <div
              className="MuiStack-root css-niqf4j-MuiStack-root"
            />
          </div>
        </div>
      </div>
    </div>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1qpzhgb-MuiGrid-root{box-sizing:border-box;margin:0;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-flex-basis:75%;-ms-flex-preferred-size:75%;flex-basis:75%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:75%;}@media (min-width:600px){.css-1qpzhgb-MuiGrid-root{-webkit-flex-basis:75%;-ms-flex-preferred-size:75%;flex-basis:75%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:75%;}}@media (min-width:900px){.css-1qpzhgb-MuiGrid-root{-webkit-flex-basis:75%;-ms-flex-preferred-size:75%;flex-basis:75%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:75%;}}@media (min-width:1200px){.css-1qpzhgb-MuiGrid-root{-webkit-flex-basis:75%;-ms-flex-preferred-size:75%;flex-basis:75%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:75%;}}@media (min-width:1536px){.css-1qpzhgb-MuiGrid-root{-webkit-flex-basis:75%;-ms-flex-preferred-size:75%;flex-basis:75%;-webkit-box-flex:0;-webkit-flex-grow:0;-ms-flex-positive:0;flex-grow:0;max-width:75%;}}",
        }
      }
      data-emotion="css 1qpzhgb-MuiGrid-root"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
    >
      <div
        className="shark-sales-chart"
      >
        <img
          alt="a chart of shark sales over time"
          className=""
          src="chart.png"
        />
      </div>
      <div
        className="shark-sales"
      >
        <div
          className="sales-graph"
        />
        <div
          className=""
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-rorn0c-MuiTableContainer-root{width:100%;overflow-x:auto;}",
              }
            }
            data-emotion="css rorn0c-MuiTableContainer-root"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-11xur9t-MuiPaper-root-MuiTableContainer-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);width:100%;overflow-x:auto;}",
              }
            }
            data-emotion="css 11xur9t-MuiPaper-root-MuiTableContainer-root"
          />
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-quj9j6-MuiTable-root{display:table;width:100%;border-collapse:collapse;border-spacing:0;min-width:650px;}.css-quj9j6-MuiTable-root caption{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;padding:16px;color:rgba(0, 0, 0, 0.6);text-align:left;caption-side:bottom;}",
                }
              }
              data-emotion="css quj9j6-MuiTable-root"
            />
            <table
              aria-label="a table view of all sales for Loading Product Details product"
              className="MuiTable-root css-quj9j6-MuiTable-root"
              role={null}
            >
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-15wwp11-MuiTableHead-root{display:table-header-group;}",
                  }
                }
                data-emotion="css 15wwp11-MuiTableHead-root"
              />
              <thead
                className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                role={null}
              >
                <style
                  dangerouslySetInnerHTML={
                    {
                      "__html": ".css-1q1u3t4-MuiTableRow-root{color:inherit;display:table-row;vertical-align:middle;outline:0;}.css-1q1u3t4-MuiTableRow-root.MuiTableRow-hover:hover{background-color:rgba(0, 0, 0, 0.04);}.css-1q1u3t4-MuiTableRow-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1q1u3t4-MuiTableRow-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}",
                    }
                  }
                  data-emotion="css 1q1u3t4-MuiTableRow-root"
                />
                <tr
                  className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <style
                    dangerouslySetInnerHTML={
                      {
                        "__html": ".css-1howxi1-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:left;padding:6px 16px;color:rgba(0, 0, 0, 0.87);}.css-1howxi1-MuiTableCell-root.MuiTableCell-paddingCheckbox{width:24px;padding:0 12px 0 16px;}.css-1howxi1-MuiTableCell-root.MuiTableCell-paddingCheckbox>*{padding:0;}",
                      }
                    }
                    data-emotion="css 1howxi1-MuiTableCell-root"
                  />
                  <th
                    aria-sort="ascending"
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                    scope="col"
                  >
                    <style
                      dangerouslySetInnerHTML={
                        {
                          "__html": ".css-o11grf-MuiTableSortLabel-root{cursor:pointer;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-flex-direction:inherit;-ms-flex-direction:inherit;flex-direction:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-o11grf-MuiTableSortLabel-root:focus{color:rgba(0, 0, 0, 0.6);}.css-o11grf-MuiTableSortLabel-root:hover{color:rgba(0, 0, 0, 0.6);}.css-o11grf-MuiTableSortLabel-root:hover .MuiTableSortLabel-icon{opacity:0.5;}.css-o11grf-MuiTableSortLabel-root.Mui-active{color:rgba(0, 0, 0, 0.87);}.css-o11grf-MuiTableSortLabel-root.Mui-active .MuiTableSortLabel-icon{opacity:1;color:rgba(0, 0, 0, 0.6);}",
                        }
                      }
                      data-emotion="css o11grf-MuiTableSortLabel-root"
                    />
                    <style
                      dangerouslySetInnerHTML={
                        {
                          "__html": ".css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;cursor:pointer;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-flex-direction:inherit;-ms-flex-direction:inherit;flex-direction:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root::-moz-focus-inner{border-style:none;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:focus{color:rgba(0, 0, 0, 0.6);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:hover{color:rgba(0, 0, 0, 0.6);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:hover .MuiTableSortLabel-icon{opacity:0.5;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-active{color:rgba(0, 0, 0, 0.87);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-active .MuiTableSortLabel-icon{opacity:1;color:rgba(0, 0, 0, 0.6);}",
                        }
                      }
                      data-emotion="css 1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                    />
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      WEEK ENDING
                      <style
                        dangerouslySetInnerHTML={
                          {
                            "__html": ".css-ty8jgw{border:0px solid;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0px;position:absolute;white-space:nowrap;width:1px;}",
                          }
                        }
                        data-emotion="css ty8jgw"
                      />
                      <span
                        className="MuiBox-root css-ty8jgw"
                      >
                        sorted ascending
                      </span>
                      <style
                        dangerouslySetInnerHTML={
                          {
                            "__html": ".css-1hi7cb7-MuiTableSortLabel-icon{font-size:18px;margin-right:4px;margin-left:4px;opacity:0;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-transform:rotate(180deg);-moz-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
                          }
                        }
                        data-emotion="css 1hi7cb7-MuiTableSortLabel-icon"
                      />
                      <style
                        dangerouslySetInnerHTML={
                          {
                            "__html": ".css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;font-size:18px;margin-right:4px;margin-left:4px;opacity:0;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-transform:rotate(180deg);-moz-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
                          }
                        }
                        data-emotion="css 1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                      />
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                        data-testid="ArrowDownwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <style
                    dangerouslySetInnerHTML={
                      {
                        "__html": ".css-6qfsqn-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:right;padding:6px 16px;color:rgba(0, 0, 0, 0.87);-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;}.css-6qfsqn-MuiTableCell-root.MuiTableCell-paddingCheckbox{width:24px;padding:0 12px 0 16px;}.css-6qfsqn-MuiTableCell-root.MuiTableCell-paddingCheckbox>*{padding:0;}",
                      }
                    }
                    data-emotion="css 6qfsqn-MuiTableCell-root"
                  />
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      RETAIL SALES
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                        data-testid="ArrowDownwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      WHOLESALE SALES
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                        data-testid="ArrowDownwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      UNITS SOLD
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                        data-testid="ArrowDownwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                    scope="col"
                  >
                    <span
                      className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      RETAILER MARGIN
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                        data-testid="ArrowDownwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                        />
                      </svg>
                    </span>
                  </th>
                </tr>
              </thead>
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-apqrd9-MuiTableBody-root{display:table-row-group;}",
                  }
                }
                data-emotion="css apqrd9-MuiTableBody-root"
              />
              <tbody
                className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                role={null}
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
